core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -681,8 +681,8 @@ package object matryoshka {
     *
     * @group algtrans
     */
-  @tailrec
```

Comment:
I would like this to be a function, but the problem is that we need this to be tail recursive.

We could actually rewrite this as a coalgebra, like:
```scala
final def repeatedlyƒ[A](f: A => Option[A]): Coalgebra[A \/ ?, A] = f(expr) \/> expr

final def repeatedly[A](f: A => Option[A]): A => A =
  _.ana[Nu](repeatedlyƒ(f)).unsafePerformSync
```
We have some stack-safety tests of `Partial` that make me think this works, although no performance promises.

But then maybe we can rewrite stuff to use `repeatedlyƒ` instead of `repeatedly`, and track partiality through more of the code.

@edmundnoble – thoughts?